tests/dynamic_data/users_roles/delete_user_test.py

```diff
@@ -1,47 +1,97 @@
 import pytest
 from layman import app, util as layman_util
-from test_tools import process_client
+from test_tools import process_client, role_service, process
 
 
-USERNAME = 'test_delete_user'
+@pytest.fixture
+def _setup_env_vars():
+    env_vars = dict(process.AUTHN_SETTINGS)
+    env_vars['GRANT_DELETE_OTHER_USER'] = "test_delete_user2,ADMIN"
+    process.ensure_layman_function(env_vars)
 
 
-@pytest.mark.usefixtures('ensure_layman_module', 'oauth2_provider_mock')
-def test_delete_user():
-    process_client.reserve_username(USERNAME, actor_name=USERNAME)
+@pytest.fixture
+def setup_users_and_role(request, _setup_env_vars):
+    username, actor_name, rolename = request.param
+    process_client.reserve_username(username, actor_name=username)
+    if actor_name != username:
+        with app.app_context():
+            usernames = layman_util.get_usernames(use_cache=False)
+            if actor_name not in usernames:
+                process_client.reserve_username(actor_name, actor_name=actor_name)
```

Comment:
These four lines can be replaced with `process_client.ensure_reserved_username(actor_name)`